src/member.rs

```diff
@@ -185,13 +185,14 @@ where
             .expect("Channel to consensus should be open")
     }
 
-    async fn on_create(&mut self, u: PreUnit<H>) {
+    async fn on_create(&mut self, u: PreUnit<H>, parent_hashes: Vec<H::Hash>) {
         debug!(target: "AlephBFT-member", "{:?} On create notification.", self.index());
         let data = self.data_io.get_data();
         let full_unit = FullUnit::new(u, data, self.config.session_id);
         let hash = full_unit.hash();
         let signed_unit = Signed::sign(full_unit, self.keybox).await;
         self.store.add_unit(signed_unit, false);
+        self.store.add_parents(hash, parent_hashes);
```

Comment:
Maybe you could add a comment here (or some other places) highlighting the importance of this change. There is a task for optimizing how we store parents and it would be nice to make sure we won't reintroduce the bug again.

Reply:
Right, I added a comment clarifying this.